docker/package/package_generator.py

```diff
@@ -132,6 +132,12 @@ def main():
 
     if target_os == "ubuntu":
         for package in packages:
+            source_archive = f"{package.name}_{package.meta.version}.orig.tar.gz"
+            print(source_archive)
```

Comment:
Is this intentional or left behind?

On a related note: I was thinking that perhaps we should throw an error in case this option is supplied but there is no source in the directory for the package that we are building.

At the moment AFAICT if there is no source package the build happens as usual, which is potentially very confusing IMO because it's not possible to notice the issue (before the package gets rejected by launchpad).

P.s. I don't think that a warning or a log message would be sufficient for this because the build produces too much output and as a consequence the message would easily get lost.

Reply:
I believe it could be inconvenient. We never need source archive for `tezos-baking`. So if we go your way, we need manually specify to build all packages except `tezos-baking`, and then `tezos-baking` separately. Now, we can just put all source archives under `sources-dir` and run `./docker/docker-tezos-packages.py --os ubuntu --type source --sources-dir sources` and everything will be done. The only source of mistake here is downloading source archive with wrong version.

Reply:
We could output warning about it, or actually we can automate downloading source archives from launchpad, which sounds like a good idea

Reply:
> we can automate downloading source archives from launchpad

Yep, I think this is a good idea as well, the URL are hopefully predictable and can be used.

However I would do this in a separate issue/PR and moreover I'd leave the possibility to specify a `sources-dir`, as there might be different situations in which we don't want to use the latest uploaded sources.

Reply:
> We never need source archive for tezos-baking.

Can't we instead make an exception for it, then?

That could be the only package (as it's the only one with non-repeating sources for which we use the `letter_version` system) where an error isn't raised if the previous source is not in the provided `--sources-dir`